odcockpit/src/plugins/environmentviewer/EnvironmentViewerGLWidget.cpp

```diff
@@ -421,70 +529,85 @@ namespace cockpit {
+                                    } else {
+                                        drawOneCPCPointWithIntensity(distance_integer, azimuth, m_12_verticalAngles[sensorIndex], distanceEncoding, numberOfBitsForIntensity, intensityPlacement, tmpMask, intensityMaxValue);
+                                    }
                                 }
-                                
-                                switch (distanceEncoding) {
-                                    case CompactPointCloud::CM : distance = static_cast<float>(cappedDistance / 100.0f); //convert to meter from resolution 1 cm
-                                                                 break;
-                                    case CompactPointCloud::MM : distance = static_cast<float>(cappedDistance / 500.0f); //convert to meter from resolution 2mm
-                                                                 break;
-                                } 
+                                azimuth += azimuthIncrement;
                             }
-                           
-                            if (distance > 1.0f) {//Only viualize the point when the distance is larger than 1m
-                                // Compute x, y, z coordinate based on distance, azimuth, and vertical angle
-                                xyDistance = distance * cos(verticalAngle * static_cast<float>(cartesian::Constants::DEG2RAD));
-                                xData = xyDistance * sin(azimuth * static_cast<float>(cartesian::Constants::DEG2RAD));
-                                yData = xyDistance * cos(azimuth * static_cast<float>(cartesian::Constants::DEG2RAD));
-                                zData = distance * sin(verticalAngle * static_cast<float>(cartesian::Constants::DEG2RAD));
-                                if(numberOfBitsForIntensity != 0) {
-                                    //The number of intensity levels depends on number of bits for intensity. There are 2^n intensity levels for n bits
-                                    float intensityLevel = intensity / intensityMaxValue;
-                                    //Four color levels: blue, green, yellow, red from low intensity to high intensity
-                                    if (intensityLevel < 0.25f + 1e-7) {
-                                        glColor3f(0.0f, 0.5f + intensityLevel * 2.0f, 1.0f);
-                                    } else if (intensityLevel > 0.25f && intensityLevel < 0.5f + 1e-7) {
-                                        glColor3f(0.0f, 0.5f + intensityLevel * 2.0f, 0.5f);
-                                    } else if (intensityLevel > 0.5f && intensityLevel < 0.75f + 1e-7) {
-                                        glColor3f(1.0f, 0.75f + intensityLevel, 0.0f);
+                        }
+                        if ((m_cpcMask_32 & 0x02) > 0) {//The second part, 11 layers
```

Comment:
These three if statements contain very many duplicated code; can you refactor this code to improve its maintainability?